src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,315 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.TextWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP
+namespace Microsoft.OData.Json
+{
+    using System;
+    using System.Buffers;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A TextWriter for writing JSON data.</returns>
+        public TextWriter StartTextWriterValueScope(string contentType)
+        {
+            this.WriteSeparatorIfNecessary();
+            this.currentContentType = contentType;
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            this.Flush();
+
+            return new ODataUtf8JsonTextWriter(this);
+        }
+
+        /// <summary>
+        /// Ends a TextWriter value scope.
+        /// </summary>
+        public void EndTextWriterValueScope()
+        {

```

Comment:
How come we don't call `textWriter.Flush()` or `textWriter.Dispose` when we end the scope? What if the text writer still has some pending unprocessed bytes from previous chunk? Those won't be written? If that's the case, that would be a bug. And if that's bug, it means there's a gap in the tests as well since that should ideally be captured by tests. I would suggest adding tests for such scenarios, i.e. where the last write leaves unprocessed bytes before you call EndScope.

For base64 that would be straightforward, I think content lengths not divisible by 3 would leave pending bytes. For text, I'm not sure. If a write call ends on character that is half of a surrogate pair (e.g. emojis take two `char`s to represent one visible character), then it may end up being unprocessed and buffered for the next write. See: https://learn.microsoft.com/en-us/dotnet/standard/base-types/character-encoding-introduction

Reply:
I think this scenario for unprocessed bytes has been covered. I've added logic for disposing the text writer from the end scope.. I think in the tests it was the caller that was calling the dispose or flush functions..

Reply:
Any write that is not a finalBlock will always have unprocessed bytes.